Volume/Views/MainView/MagazineReaderView.swift

```diff
@@ -0,0 +1,222 @@
+//
+//  MagazineDetail.swift
+//  Volume
+//
+//  Created by Justin Ngai on 9/3/2022.
+//  Copyright © 2022 Cornell AppDev. All rights reserved.
+//
+
+import SwiftUI
+import PDFKit
+import AppDevAnalytics
+import Combine
+import LinkPresentation
+import SDWebImageSwiftUI
+
+struct MagazineReaderView: View {
+    @Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
+    @EnvironmentObject private var userData: UserData
+    
+    let magazine: Magazine
+    let navigationSource: NavigationSource
+    @State private var bookmarkRequestInProgress = false
+    @State private var cancellableShoutoutMutation: AnyCancellable?
+    @State private var showToolbars = true
+    
+    private var isShoutoutsButtonEnabled: Bool {
+        return userData.canIncrementMagazineShoutouts(magazine)
+    }
+    
+    private var navbar: some View {
+            ZStack {
+                Rectangle()
+                    .foregroundColor(.white)
+                    .shadow(color: .black.opacity(0.2), radius: 2, y: 3)
+                
+                if showToolbars {
+                    HStack {
+                        Button {
+                            presentationMode.wrappedValue.dismiss()
+                        } label: {
+                            Image.volume.leftArrow
+                                .resizable()
+                                .aspectRatio(contentMode: .fit)
+                                .frame(height: 24)
+                                .foregroundColor(.black)
+                        }
+                        
+                        Spacer()
+                        
+                        if let url = magazine.pdfUrl {
+                            Link(destination: url) {
+                                Image.volume.compass
+                                    .resizable()
+                                    .aspectRatio(contentMode: .fit)
+                                    .frame(height: 24)
+                                    .foregroundColor(.black)
+                            }
+                        } else {
+                            Image.volume.compass
+                                .resizable()
+                                .aspectRatio(contentMode: .fit)
+                                .frame(height: 24)
+                                .foregroundColor(.volume.lightGray)
+                        }
+                    }
+                    .padding(.horizontal, 16)
+                    
+                    VStack {
+                        Text(magazine.title)
+                            .font(.newYorkBold(size: 12))
+                            .truncationMode(.tail)
+                        Text("Reading in Volume")
+                            .font(.helveticaRegular(size: 10))
+                            .foregroundColor(.volume.lightGray)
+                    }
+                    .padding(.horizontal, 48)
```

Comment:
You may want to consider factoring this out into its own variable. The `navBar` property is getting pretty large.